pkg/jwt/forwarding_signature_validator.go

```diff
@@ -0,0 +1,26 @@
+package jwt
+
+import (
+	"sync/atomic"
+)
+
+type forwardingSignatureValidator struct {
+	validator atomic.Pointer[SignatureValidator]
+}
+
+// NewForwardingSignatureValidator creates a SignatureValidator that simply forwards
+// requests to another SignatureValidator.
+func NewForwardingSignatureValidator(validator SignatureValidator) SignatureValidator {
```

Comment:
Any reason to not return the concrete struct here? Common go pattern is to "Accept interfaces, return structs"

Reply:
It's the pattern they've adopted, so I'm forced to follow that I think